samcli/lib/pipeline/bootstrap/resource.py

```diff
@@ -0,0 +1,133 @@
+""" Represents AWS resource"""
+from typing import List, Optional
+
+
+class ARNParts:
+    """
+    Decompose a given ARN into its parts https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
+
+    Attributes
+    ----------
+    partition: str
+        the partition part(AWS, aws-cn or aws-us-gov)  of the ARN
+    service: str
+        the service part(S3, IAM, ECR, ...etc) of the ARN
+    region: str
+        the AWS region part(us-east-1, eu-west-1, ...etc) of the ARN
+    account-id: str
+        the account-id part of the ARN
+    resource-id: str
+        the resource-id part of the ARN
+    resource-type: str
+        the resource-type part of the ARN
+    """
+
+    def __init__(self, arn: str) -> None:
+        if not isinstance(arn, str):
+            raise ValueError(f"Invalid ARN ({arn}) is not a String")
+
+        parts = arn.split(":")
+        try:
+            self.partition: str = parts[1]
+            self.service: str = parts[2]
+            self.region: str = parts[3]
+            self.account_id: str = parts[4]
+            self.resource_id: str = parts[5]
+            self.resource_type: Optional[str] = None
+        except IndexError as ex:
+            raise ValueError(f"Invalid ARN ({arn})") from ex
+
+        if "/" in self.resource_id:
+            resource_type_and_id: List[str] = self.resource_id.split("/")
+            self.resource_type = resource_type_and_id[0]
+            self.resource_id = resource_type_and_id[1]
```

Comment:
simpler version: 

```suggestion
            [self.resource_type, self.resource_id] = self.resource_id.split("/")
```

Reply:
actually, I'm reverting this, figured out resource_id can have multiple slashes (resource_path)